indico/modules/events/contributions/util.py

```diff
@@ -221,6 +221,25 @@ def has_contributions_with_user_as_submitter(event, user):
     return _query_contributions_with_user_as_submitter(event, user).has_rows()
 
 
+def get_contributions_for_person(event, person, onlyspeakers=False):
+    """Get all contributions for an event person
+
+    If onlyspeakers is true, then only contributions where the person is speaker
+    are returned
+    """
+    if onlyspeakers:
+        cl_join = db.and_(ContributionPersonLink.is_speaker,
+                          ContributionPersonLink.person_id == person.id,
+                          ContributionPersonLink.contribution_id == Contribution.id)
+    else:
+        cl_join = db.and_(ContributionPersonLink.person_id == person.id,
+                          ContributionPersonLink.contribution_id == Contribution.id)
+
+    return (Contribution.query.filter(Contribution.event_id == event.id)
```

Comment:
`Contribution.query.with_parent(event)` queries contributions for the given event, and already filters out deleted ones, so you can completely skip the filter in the line below.

Reply:
Done